### PR TITLE
docs: Update the feature deprecation page

### PR DESCRIPTION
**Pull request by mpalmi**
Update the feature deprecation page to describe the deprecation statuses and their effects.

### CODE REVIEW

File: website/content/docs/deprecation/index.mdx [hunk 1]
```diff
@@ -10,6 +10,45 @@ description: |-
...
+This status reflects a feature which has been marked for deprecation in a later release of Vault. This is the first phase of the deprecation process. A status of `Deprecated` has two effects:
+
+1) All attempts to utilize this feature will result in `Warn`-level messages in the Vault log.
+2) All attempts to access an endpoint which supports this feature will result in `warnings` in the Response.
+
+### Pending Removal
+
+This status reflects a feature which has been officially deprecated in this release of Vault. This is the first phase in the process that fundamentally alters the behavior of Vault. The effects are two-fold:
+
+1) After an upgrade, any existing `Pending Removal` mount will cause Vault to log an `Error`-level message in the Vault log and automatically shutdown.
```
Comment (ncabatoff) [inline website/content/docs/deprecation/index.mdx:36]: Up until this point, all the text has been speaking about deprecation status of *features* and and *endpoints*.  Now suddenly we're speaking about it solely in terms of mounts.  Maybe we should keep the focus on mounts throughout?

Comment (mladlow) [reply website/content/docs/deprecation/index.mdx]: I'd hazard a guess that focusing on features would be more user friendly than something like "mounts" which is kind of second-level terminology.

Comment (mpalmi) [reply website/content/docs/deprecation/index.mdx]: Thank you both. I can make the wording more consistent around "feature" as opposed to "mount".

File: website/content/docs/deprecation/index.mdx [hunk 2]
```diff
@@ -10,6 +10,60 @@ description: |-
 
 This announcement page is maintained and updated periodically to communicate important decisions made concerning End of Support(EoS) for Vault features as well as features we have removed or disabled from the product. We document the removal of features, enable the community with a plan and timeline for eventual deprecations, and supply alternative paths to explore and evaluate to minimize business disruptions. If you have questions or concerns about a deprecated feature, please create a topic on [the community forum](https://discuss.hashicorp.com/c/vault/30) or raise a ticket with your support team. Please refer to the [FAQ](/docs/deprecation/faq) page for frequently asked questions concerning Vault feature deprecations.
 
+## Deprecation Status
```
Comment (mladlow) [inline website/content/docs/deprecation/index.mdx:13]: Thanks for updating the wording. I think the concept of mount is still probably confusing but I imagine it's a point of confusion across our docs, so I dunno how important it is to tidy it here.

I think this section should probably either go on the FAQ page or _under_ the deprecation table on this page, but open to other's opinions on this, and ok with the wording, so I'm going to approve it.

**Review by mladlow — APPROVED** [review]

**Pull request merged**